src/TableOfContents/index.tsx

```diff
@@ -42,27 +42,27 @@ export type ITableOfContentsLink = TableOfContentsItem & {
   isExternalLink?: boolean;
 };
 
-export type RowComponentProps<T extends TableOfContentsItem> = {
+export type RowComponentProps<T extends TableOfContentsItem, E = {}> = {
   item: T;
   index: number;
   isExpanded: boolean;
   toggleExpanded: () => void;
-};
+} & ({} extends E ? {} : { extra: E });
 
-export type RowComponentType<T extends TableOfContentsItem> = React.ComponentType<RowComponentProps<T>>;
+export type RowComponentType<T extends TableOfContentsItem, E = {}> = React.ComponentType<RowComponentProps<T, E>>;
 
-export interface ITableOfContents<T extends TableOfContentsItem = TableOfContentsItem> {
+export type ITableOfContents<T extends TableOfContentsItem = TableOfContentsItem, E = {}> = {
```

Comment:
That should probably be renamed to `TableOfContentsType` but that would be a breaking change so I guess better keep it for now.

Reply:
You're right. I'm not a great fan of the `I` prefix no matter if it's an interface or a type (there's barely any difference anymore) because IMO it brings no value and generally advised against in official TS sources (just google "typescript i prefix").

I'd really just name it `TableOfContentsProps`, that seems to be the convention in most libs, but as you guessed I didn't want to make a breaking change here.